qiskit/algorithms/linear_solvers/matrices/tridiagonal.py

```diff
@@ -0,0 +1,210 @@
+            super().__init__(qr_state, qr_ancilla, name=name)
+        else:
+            super().__init__(qr_state, name=name)
+
+        self._num_state_qubits = None
+
+        self._main_diag = main_diag
+        self._off_diag = off_diag
+        self._tolerance = tolerance if tolerance is not None else 1e-2
+        self._time = time if time is not None else 1
+        self._trotter = trotter
+
+        self._num_state_qubits = num_state_qubits
+
+    def set_simulation_params(self, time: float, tolerance: float):
+        """Override the simulation parameters"""
+        self.tolerance(tolerance)
+        self.time(time)
+
+    def tolerance(self, tolerance: float):
+        """Return the error tolerance"""
+        self._tolerance = tolerance
+
+    def time(self, time: float):
+        """Return the time of the evolution"""
+        self._time = time
+        # Update the number of trotter steps. Max 7 for now, upper bounds too loose.
+        self._trotter = min(self._num_state_qubits + 1,
+                            int(np.ceil(np.sqrt(((time * np.abs(self._off_diag)) ** 3)
+                                                / 2 / self._tolerance))))
+
+    def matrix(self) -> np.ndarray:
+        """Return the matrix"""
+        matrix = diags([self._off_diag, self._main_diag, self._off_diag], [-1, 0, 1],
+                       shape=(2 ** self._num_state_qubits, 2 ** self._num_state_qubits)).toarray()
+        return matrix
+
+    def _cn_gate(self, qc: QuantumCircuit, controls: QuantumRegister, qr_a: AncillaRegister,
+                 phi: float, ulambda: float, theta: float, tgt: Qubit) -> QuantumCircuit:
+        """Apply an n-controlled gate.
+
+        Args:
+            qc: Thq QuantumCircuit implementing the unitary gate
+            controls: list of controlling qubits
+            qr_a: ancilla register
+            phi: argument for a general qiskit u gate
+            ulambda: argument for a general qiskit u gate
+            theta: argument for a general qiskit u gate
+            tgt: target qubit
+
+        Returns:
+            The quantum circuit implementing a multi-controlled unitary.
+        """
+        # The first Toffoli
+        qc.ccx(controls[0], controls[1], qr_a[0])
+        for i in range(2, len(controls)):
+            qc.ccx(controls[i], qr_a[i - 2], qr_a[i - 1])
+        # Now apply the 1-controlled version of the gate with control the last ancilla bit
+        qc.cu(theta, phi, ulambda, 0, qr_a[len(controls) - 2], tgt)
+
+        # Uncompute ancillae
+        for i in range(len(controls) - 1, 1, -1):
+            qc.ccx(controls[i], qr_a[i - 2], qr_a[i - 1])
+        qc.ccx(controls[0], controls[1], qr_a[0])
+        return qc
+
+    # Controlled version of the circuit for the main diagonal
+    def _build_main_controlled(self, qc: QuantumCircuit, q_control: Qubit,
+                               params: Optional[float] = 1) -> QuantumCircuit:
```

Comment:
Similar comment as trotter constructor param over Optional - this should just be `params: float = 1`. And can we change the name - params could mean anything. Where this is passed to the gate the parameter there is called `theta`. I think it would be better to be callled `theta` here too

Reply:
Removed Optional and renamed `params` to `theta`.